src/Microsoft.Health.Fhir.Liquid.Converter.Tool/TemplateManagementLogicHandler.cs

```
@@ -0,0 +1,89 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Fhir.Liquid.Converter.Tool.Models;
+using Microsoft.Health.Fhir.TemplateManagement;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Liquid.Converter.Tool
+{
+    internal static class TemplateManagementLogicHandler
+    {
+        private static readonly ILogger Logger;
+        private static readonly TextWriter ErrorWriter;
+
+        static TemplateManagementLogicHandler()
+        {
+            using var loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder.AddFilter("Microsoft.Health.Fhir.TemplateManagement", LogLevel.Trace)
+                       .AddConsole();
+            });
+            FhirConverterLogging.LoggerFactory = loggerFactory;
+            Logger = FhirConverterLogging.CreateLogger(typeof(TemplateManagementLogicHandler));
+
+            ErrorWriter = Console.Error;
+        }
+
+        internal static async Task PullAsync(PullTemplateOptions options)
+        {
+            try
+            {
+                if (!options.ForceOverride)
+                {
+                    if (Directory.Exists(options.OutputTemplateFolder) && Directory.GetFileSystemEntries(options.OutputTemplateFolder).Length != 0)
+                    {
+                        Logger.LogError("The output folder is not empty. If force to override, please add -f in parameters");
```

If the folder is not empty, we can throw an exception.